app/components/polls/polls-edit/polls-edit-directive.js

```diff
@@ -16,60 +16,113 @@ angular.module('polls.edit.directive', [
     value: 'Image'
   }
 ])
-  .directive('pollsEdit', function (routes) {
-    return {
-      templateUrl: routes.COMPONENTS_URL + 'polls/polls-edit/polls-edit.html',
-      controller: function (_, $location, $q, $routeParams, $scope, $window, Poll) {
-        // populate model for use
-        if ($routeParams.id === 'new') {
-          $scope.model = Poll.$build();
-          $scope.isNew = true;
-        } else {
-          $scope.model = Poll.$find($routeParams.id);
+.directive('pollsEdit', function (routes) {
+  return {
+    templateUrl: routes.COMPONENTS_URL + 'polls/polls-edit/polls-edit.html',
+    controller: function (_, $http, $location, $q, $routeParams, $scope, Poll) {
+      // populate model for use
+      if ($routeParams.id === 'new') {
+        $scope.model = Poll.$build();
+        $scope.isNew = true;
+      } else {
+        $scope.model = Poll.$find($routeParams.id);
+        $scope.poll = Poll.$find($routeParams.id);
+      }
+
+      window.onbeforeunload = function (e) {
+        if(!_.isEmpty($scope.model.$dirty()) || $scope.isNew || $scope.needsSave) {
+          // show confirmation alert
+          return 'You have unsaved changes.';
         }
+      };
 
-        window.onbeforeunload = function (e) {
-          if(!_.isEmpty($scope.model.$dirty()) || $scope.isNew || $scope.needsSave) {
-            // show confirmation alert
-            return 'You have unsaved changes.';
-          }
-        };
+      $scope.$on('$destroy', function () {
+        // remove alert when we go
+        delete window.onbeforeunload;
+      });
 
-        $scope.$on('$destroy', function () {
-          // remove alert when we go
-          delete window.onbeforeunload;
-        });
+      var answerUrl = '/cms/api/v1/answer/';
 
       $scope.saveModel = function () {
         if ($scope.model) {
+          $scope.deleteSodaheadAnswers();
+
+          _.forEach($scope.model.answers, function(answer) {
+            if(!$scope.isNew && !answer.notOnSodahead) {
+              $scope.putSodaheadAnswer(answer);
+            }
+            if(!$scope.isNew && answer.notOnSodahead) {
+              $scope.postSodaheadAnswer(answer, $scope.model.$pk);
+            }
+          });
+
+          $scope.answers = $scope.model.answers;
           return $scope.model.$save().$asPromise().then(function (data) {
-            $location.path('/cms/app/polls/edit/' + data.id + '/');
+            if($scope.isNew) {
+              var answerPromises = _.map($scope.answers, function (answer) {
+                return $scope.postSodaheadAnswer(answer, data.id);
```

Comment:
Is this actually posting to our API, instead of Sodaheads?  Thought Sodahead allowed you to post everything all at once